pkg/util/provider/drain/drain.go

```diff
@@ -232,20 +239,26 @@ func (o *Options) RunDrain(ctx context.Context) error {
 		klog.Errorf("Drain Error: Cordoning of node failed with error: %v", err)
 		return err
 	}
+	stopCh := make(chan struct{})
```

Comment:
We should not construct an empty `stopCh` here as it becomes useless for signalling. We should use a channel obtained from `context.WithTimeout` using the drain timeout passed to `NewDrainOptions`. And then use the `Done()` method on the returned context to get the `stopCh`.

Reply:
PS: There is already a context created for this a little later in `RunDrain`. `	drainContext, cancelFn := context.WithDeadline(ctx, o.drainStartedOn.Add(o.Timeout))
`

We should use the `Done()` channel from this.